drivers/mrf24j40/mrf24j40.c

```diff
@@ -157,9 +157,13 @@ void mrf24j40_tx_exec(mrf24j40_t *dev)
     mrf24j40_reg_write_long(dev, MRF24J40_TX_NORMAL_FIFO, dev->header_len);
 
     if (dev->netdev.flags & NETDEV_IEEE802154_ACK_REQ) {
-        mrf24j40_reg_write_short(dev, MRF24J40_REG_TXNCON, MRF24J40_TXNCON_TXNACKREQ | MRF24J40_TXNCON_TXNTRIG);
+        /* Explicitly set the TXNACKREQ register before sending */
+        mrf24j40_reg_write_short(dev, MRF24J40_REG_TXNCON, MRF24J40_TXNCON_TXNACKREQ);
+        mrf24j40_reg_write_short(dev, MRF24J40_REG_TXNCON, MRF24J40_TXNCON_TXNTRIG);
```

Comment:
btw, you are clearing the TXNACKREQ flag here, is that intentional?

Reply:
> Ah, I misunderstood the original problem statement. So the hardware triggers the TX before the other bit settings take effect.

I think that's exactly what happens. The TX is triggered with the previous state of the TXNCON register. Thus settings applied to the TXNCON register in the same SPI transfer apply to the next transmission (I think). This stays speculation about the inner working of the mrf24j40.
> Basically I interpret that as they want you to set all configuration bits first while loading the frame, then trigger the TX when everything is ready.

Yeah, when interpreting the datasheet literally, that's what should be done